vllm/core/scheduler.py

```diff
@@ -1031,17 +1084,35 @@ def _can_append_slots(self, seq_group: SequenceGroup) -> bool:
             num_lookahead_slots=self._get_num_lookahead_slots(is_prefill),
         )
 
-    def schedule(self) -> Tuple[List[SequenceGroupMetadata], SchedulerOutputs]:
+    def _allow_async_output_proc(self, seq_group: SequenceGroup) -> bool:
+        no_beam_search = (seq_group.sampling_params.n == 1
+                          and not seq_group.sampling_params.use_beam_search)
```

Comment:
maybe this?
```suggestion
    return seq_group.best_of == 1
```

Reply:
I have copy-pasted the original check from the single-step to be on the safe.

Reply:
1. Parallel sampling also requires disabling this feature, right?
2. `n` means the number of output sequences to return, while `best_of` means the number of sequences that are actually generated. Therefore, `best_of` is the right parameter to check whether the request uses beam search or parallel sampling.

Reply:
Submitted a PR to fix it in the single-step #7881

Reply:
Oh I see, changed the condition here to be in sync with https://github.com/vllm-project/vllm/pull/7881